apollo-router/tests/integration/oci.rs

```diff
@@ -22,7 +22,7 @@ use crate::integration::common::graph_os_enabled;
 
```

Comment:
can we beef out this file a bit with integration tests for:

* making sure tags are handled correctly
* hot reloading works as expected (looks like they're all set to `false`)
* what happens to HEAD requests when a 500/404/403/401 gets returned--mostly that the last good artifact is still in use and that we can survive registry being intermittently unavailable
* it'd be great to see what happens on a bad artifact via an integration test

Reply:
We have some unit tests around the hot reload case (e.g. stream_from_oci_digest_changed_fetches_schema() ), but I can add an integration test as well (both for the happy path case and for some of the unhappy path tests).

Reply:
Added integration tests!